### PR TITLE
Shade Jackson into a safe location for use of WorldWind as a library

### PR DESCRIPTION
**Pull request by roxchkplusony**

### CODE REVIEW

File: src/gov/nasa/worldwind/data/BILRasterWriter.java [hunk 1]
```diff
@@ -203,30 +202,30 @@ protected String validate(AVList worldFileParams, Object dataSource)
 
         Object o = worldFileParams.getValue(WorldFile.WORLD_FILE_IMAGE_SIZE);
         if (o == null || !(o instanceof int[]))
-            sb.append(sb.length() > 0 ? ", " : "").append(Logging.getMessage("WorldFile.NoSizeSpecified", dataSource));
+            sb.append(sb.length() > 0 ? ", " : "").append((String) null);
```
Comment (tomwang-work) [inline src/gov/nasa/worldwind/data/BILRasterWriter.java]: I think here it's probably best to just have an empty string returned.

File: src/gov/nasa/worldwind/data/GDALDataRaster.java [hunk 2]
```diff
@@ -546,18 +511,15 @@ protected Dataset getBestSuitedDataset(int reqWidth, int reqHeight, Sector reqSe
...
         }
 
         Object cs = this.getValue(AVKey.COORDINATE_SYSTEM);
         if (null == cs
             || (!AVKey.COORDINATE_SYSTEM_GEOGRAPHIC.equals(cs) && !AVKey.COORDINATE_SYSTEM_PROJECTED.equals(cs)))
         {
             String msg = (null == cs) ? "generic.UnspecifiedCoordinateSystem" : "generic.UnsupportedCoordinateSystem";
-            String reason = Logging.getMessage(msg, cs);
-            Logging.logger().finest(Logging.getMessage("generic.CannotCreateRaster", reason));
+            String reason = null;
```
Comment (tomwang-work) [inline src/gov/nasa/worldwind/data/GDALDataRaster.java]: I think msg and reason can both be removed.

File: src/gov/nasa/worldwind/data/WWDotNetLayerSetConverter.java [hunk 3]
```diff
@@ -122,24 +122,24 @@ protected String validateProductionParameters(AVList parameters)
...
         Object o = parameters.getValue(AVKey.FILE_STORE_LOCATION);
         if (o == null || !(o instanceof String) || ((String) o).length() < 1)
-            sb.append((sb.length() > 0 ? ", " : "")).append(Logging.getMessage("term.fileStoreLocation"));
+            sb.append((sb.length() > 0 ? ", " : "")).append((String) null);
 
         o = parameters.getValue(AVKey.DATA_CACHE_NAME);
         // It's okay if the cache path is empty, but if specified it must be a String.
         if (o != null && !(o instanceof String))
-            sb.append((sb.length() > 0 ? ", " : "")).append(Logging.getMessage("term.fileStoreFolder"));
+            sb.append((sb.length() > 0 ? ", " : "")).append((String) null);
```
Comment (tomwang-work) [inline src/gov/nasa/worldwind/data/WWDotNetLayerSetConverter.java]: I think you can remove everything in this method and just return null.

File: src/gov/nasa/worldwind/formats/nitfs/NITFSRuntimeException.java [hunk 4]
```diff
@@ -16,43 +14,31 @@
...
     {
-        super();
-    }
-
-    public NITFSRuntimeException(String messageID)
-    {
-        super(Logging.getMessage(messageID));
-        log(this.getMessage());
+        super((String) null);
+        log();
```
Comment (tomwang-work) [inline src/gov/nasa/worldwind/formats/nitfs/NITFSRuntimeException.java]: This call to log is an addition to this constructor. I think we want this removed.

**Comment by tomwang-work** [comment]
Just a few comments. Too many files unfortunately for GitHub to display all the diffs. But overall looks good.

File: src/gov/nasa/worldwind/formats/nitfs/NITFSRuntimeException.java (hunk 4 above)
Comment (tomwang-work) [inline src/gov/nasa/worldwind/formats/nitfs/NITFSRuntimeException.java]: N/m. I thought I saw "log" somewhere else.

**Comment by roxchkplusony** [comment]
Okay, giving up because doing it properly requires a ton more work. Just shading jackson for now

**Comment by tomwang-work** [comment]
👍

**Pull request merged**